### ISSUE TITLE
Constructing a dictionary of lambdas produces an error

### ISSUE THREAD

**Issue by crypto-rsa** [dotnet/roslyn#23758]
This example works:
```
var testMethods = new Dictionary<string, Func<int, int, bool>>()
{
    [">"] = ( curValue, testValue ) => curValue > testValue,
    ["<"] = ( curValue, testValue ) => curValue < testValue,
};
```

while the same example with swapped lines produces an error (_The variable 'int' cannot be used with type arguments. 'b' is a variable but is used like a type_):
```
var testMethods = new Dictionary<string, Func<int, int, bool>>()
{
    ["<"] = ( curValue, testValue ) => curValue < testValue,
    [">"] = ( curValue, testValue ) => curValue > testValue,
};
```

**Comment by CyrusNajmabadi** [comment]
I don't repro this behavior on 15.6 preview 1. 

Both forms work fine for me.  

--

Note: your error message is very concerning.  specifically "The variable **'int'** cannot be used with type arguments. **'b'** is a variable but is used like a type".

To me... it looks like something maybe went wrong with incremental parsing and the parser or lexer got very confused back with the ```Func<int, int, bool>>``` declaration.  The fact that it's referring to something called 'b' where there's no 'b' in your example is not good.

Tagging @gafter in case he's seen any sort of bug around this, or if any of his parser work could have fixed an issue here.

**Comment by jcouv** [comment]
I wonder if it may be related to this fix: https://github.com/dotnet/roslyn/pull/22102

**Comment by alrz** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/22856

**Comment by jcouv** [comment]
@crypto-rsa This was fixed in 15.6. I'll go ahead and close the issue.
Feel free to re-open is you're still seeing this in 15.6 preview 1 (or later). Thanks

**Comment by crypto-rsa** [comment]
@jcouv Thank you!
@CyrusNajmabadi Sorry about the error message! I actually copied it from another piece of code where I tested variables `a` and `b` but I ended up using `curValue` and `testValue` in the original comment.
